File: dataset/output/MISTRAL_codestral-latest/Task29_MISTRAL_codestral-latest.java
import java.util.HashMap;
import java.util.Map;

public class TwoSum {
    public static int[] twoSum(int[] numbers, int target) {
        Map<Integer, Integer> numMap = new HashMap<>();
        for (int i = 0; i < numbers.length; i++) {
            if (numMap.containsKey(target - numbers[i])) {
                return new int[]{numMap.get(target - numbers[i]), i};
            }
            numMap.put(numbers[i], i);
        }
        return new int[]{};
    }
}

File: dataset/output/MISTRAL_codestral-latest/Task29_MISTRAL_codestral-latest.c
#include <stdio.h>

void twoSum(int* numbers, int numbersSize, int target, int* returnSize, int* returnIndexes) {
    int numMap[10000] = {0}; // Assuming the range of numbers is within 0 to 9999
    for (int i = 0; i < numbersSize; i++) {
        if (numMap[target - numbers[i]] != 0) {
            returnIndexes[0] = numMap[target - numbers[i]] - 1;
            returnIndexes[1] = i;
            *returnSize = 2;
            return;
        }
        numMap[numbers[i]] = i + 1;
    }
    *returnSize = 0;
}